### PR TITLE
Build and Improve Auth and Listing Context

### PR DESCRIPTION
**Pull request by Pochetes**
## What happened here
- after creating create pool listing component, I decided that it would be cleaner to globalize the listings data so that I would have the ease of retrieving that data in any part of the app
- helped remove prop drilling `listings`
- created a `handlers` object for all functions in `auth` and `listings` contexts. These properties are the same api calls from `apiClient`, except it holds the validation and error checking logic. Good to separate that logic from the UI component logic.

For example, before I would make the api call in the component like this
```javascript
        const { data, error } = await apiClient.loginUser(JSON.stringify(formattedFormData))
        if (data) {
            setUser(data.user)
            apiClient.setToken(data.token)
        } if (error) {
            setError(error.message)
        }
```
By abstracting it in `useContext`, now the function call looks like this
```javascript
await authHandlers.loginUser(JSON.stringify(formattedFormData))
```

`NOTE` error checking and loading state variables are not used yet, so that is why you will see some unused returned values from hooks.
`NOTE` the `handleFullLogout` handler has `listingsHandlers.clearListings()` commented out because the listings would never show up after logging back in. 

If there is any way for me to clear out all auth, listings (and potentially reservations) data safely, please advise!

### CODE REVIEW

File: swimster-ui/src/contexts/auth.js [hunk 1]
```diff
@@ -1,13 +1,51 @@
...
+            if (data?.user) {
+                setUser(data.user)
+                apiClient.setToken(data.token)
+                localStorage.setItem("swimster_session_token", data.token)
+            } if (error) setError(error)
+        },
+        logoutUser: () => {
+            apiClient.logoutUser()
+            setUser({})
+            setInitialized(true)
```
Comment (roeshanigos) [inline swimster-ui/src/contexts/auth.js:32]: why do you setInitialized here when a user logs out?

File: swimster-ui/src/components/App/App.jsx [hunk 2]
```diff
@@ -7,64 +7,46 @@ import Navbar from '../Navbar/Navbar'
...
+        await authHandlers.fetchUserFromToken()
+        await listingsHandlers.fetchListings()
+      }
 
-      const { data, error } = await apiClient.fetchListings()
-      if (error) setError(error)
-      if (data?.listings) setListings(data.listings)
-
-      setIsFetching(false)
+      setInitialized(true)
```
Comment (roeshanigos) [inline swimster-ui/src/components/App/App.jsx:30]: you already setInitialized in fetchUserFromToken?

File: swimster-ui/src/components/Navbar/Navbar.jsx [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
 
-const Navbar = ({ handleLogout }) => {
-    const { user } = useAuthContext()
+const Navbar = () => {
+    const { handlers: authHandlers, user } = useAuthContext()
+    const { handlers: listingsHandlers } = useListingsContext()
+
+    // commented out function doesn't load listings on login after logout
+    const handleFullLogout = () => {
+        // listingsHandlers.clearListings()
```
Comment (roeshanigos) [inline swimster-ui/src/components/Navbar/Navbar.jsx:13]: unnecessary comment?

**Review by roeshanigos — APPROVED** [review]
okay so in terms of the size of this diff - i think it is good! not too big for me

minor nits

File: swimster-ui/src/contexts/auth.js (hunk 1 above)
Comment (Pochetes) [reply swimster-ui/src/contexts/auth.js:32]: Ooh, good find. I meant to set it to `false`. The `initialized` state variable is to log when all of the context from `auth`, `listings`, and `reservations` is mounted and unmounted when logging in/out

Comment (Pochetes) [reply swimster-ui/src/contexts/auth.js:32]: done!

File: swimster-ui/src/components/App/App.jsx (hunk 2 above)
Comment (Pochetes) [reply swimster-ui/src/components/App/App.jsx:30]: I did, for some reason I left it there when moving it to the `auth` context. Changed!

File: swimster-ui/src/components/Navbar/Navbar.jsx (hunk 3 above)
Comment (Pochetes) [reply swimster-ui/src/components/Navbar/Navbar.jsx:13]: done!

**Pull request merged**